<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>835</width>
    <height>883</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Protobuf Fixer</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="gridLayoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>781</width>
      <height>51</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="1">
      <widget class="QLineEdit" name="input_file_path"/>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="label">
       <property name="text">
        <string>File path</string>
       </property>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QPushButton" name="read_file_button">
       <property name="text">
        <string>Read</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>80</y>
      <width>311</width>
      <height>211</height>
     </rect>
    </property>
    <property name="title">
     <string>Header info</string>
    </property>
    <widget class="QWidget" name="formLayoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>301</width>
       <height>171</height>
      </rect>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Data type</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="data_type_control"/>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>PV name</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="pv_name_control"/>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Year</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLineEdit" name="year_control"/>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>IArray element count</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QSpinBox" name="element_count_control"/>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>370</y>
      <width>521</width>
      <height>431</height>
     </rect>
    </property>
    <property name="title">
     <string>Events</string>
    </property>
    <widget class="QTableWidget" name="events_table">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>501</width>
       <height>341</height>
      </rect>
     </property>
     <property name="columnCount">
      <number>9</number>
     </property>
     <column>
      <property name="text">
       <string>Line</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Timestamp</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Value</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Extra fields</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Errors</string>
      </property>
     </column>
     <column/>
     <column/>
     <column/>
     <column/>
    </widget>
    <widget class="QPushButton" name="move_event_up_button">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>370</y>
       <width>91</width>
       <height>36</height>
      </rect>
     </property>
     <property name="text">
      <string>Move up</string>
     </property>
    </widget>
    <widget class="QPushButton" name="move_event_down_button">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>370</y>
       <width>91</width>
       <height>36</height>
      </rect>
     </property>
     <property name="text">
      <string>Move down</string>
     </property>
    </widget>
    <widget class="QPushButton" name="delete_event_button">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>370</y>
       <width>111</width>
       <height>36</height>
      </rect>
     </property>
     <property name="text">
      <string>Delete event</string>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="decode_button">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>290</y>
      <width>301</width>
      <height>36</height>
     </rect>
    </property>
    <property name="text">
     <string>Decode events and check errors</string>
    </property>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>790</y>
      <width>321</width>
      <height>51</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout_2">
     <item row="0" column="1">
      <widget class="QLineEdit" name="save_suffix_button">
       <property name="text">
        <string>edited</string>
       </property>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QPushButton" name="save_button">
       <property name="text">
        <string>Save</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="label_8">
       <property name="text">
        <string>Save suffix</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget_3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>90</y>
      <width>201</width>
      <height>74</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout_3">
     <item row="0" column="0">
      <widget class="QLabel" name="label_9">
       <property name="text">
        <string>Status:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Header loaded</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label_7">
       <property name="text">
        <string>123 lines loaded</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>370</y>
      <width>281</width>
      <height>371</height>
     </rect>
    </property>
    <property name="title">
     <string>Errors</string>
    </property>
    <widget class="QListWidget" name="errors_list">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>20</y>
       <width>261</width>
       <height>341</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>No value for event at index 1</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Duplciated timestamp at index 11</string>
      </property>
     </item>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>835</width>
     <height>28</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
